base/environments/jax_environment.py

```diff
@@ -284,17 +284,23 @@ def save_parameters(self):
         Also saves all parameters to the data file.
         """
         from jax.util.labrad import remove_labrad_units
-
-        params = {}
-        params_full = {}
         pb = self.cxn.parameter_bank
+        params_serialized, params_full_serialized = pb.get_frozen_parameters(self.scheduler.rid)
+        params = pickle.loads(params_serialized)
+        params_full = pickle.loads(params_full_serialized)
         for collection, name in self.parameter_paths:
             if collection not in params:
```

Comment:
if collection is not in params, should it throw an error?

Reply:
I added some docstrings to clarify.